source/game.c

```diff
@@ -717,38 +717,57 @@ void set_seed(int seed)
     srand(rng_seed);
 }
 
-void sort_hand_by_suit()
+// Compare two cards for sorting by suit (primary) and rank (secondary)
+// Returns true if card_a should come before card_b
+static inline bool card_compare_by_suit(CardObject* card_a, CardObject* card_b)
 {
-    for (int a = 0; a < hand_top; a++)
-    {
-        for (int b = a + 1; b <= hand_top; b++)
-        {
-            if (hand[a] == NULL || (hand[b] != NULL && (hand[a]->card->suit > hand[b]->card->suit || (hand[a]->card->suit == hand[b]->card->suit && hand[a]->card->rank > hand[b]->card->rank))))
-            {
-                CardObject* temp = hand[a];
-                hand[a] = hand[b];
-                hand[b] = temp;
-            }
-        }
-    }
+    if (card_a == NULL) return false;
+    if (card_b == NULL) return true;
+    if (card_a->card->suit != card_b->card->suit)
+        return card_a->card->suit < card_b->card->suit;
+    return card_a->card->rank < card_b->card->rank;
+}
+
+// Compare two cards for sorting by rank only
+// Returns true if card_a should come before card_b
+static inline bool card_compare_by_rank(CardObject* card_a, CardObject* card_b)
+{
+    if (card_a == NULL) return false;
+    if (card_b == NULL) return true;
+    return card_a->card->rank < card_b->card->rank;
 }
 
-void sort_hand_by_rank()
+// Insertion sort implementation for the hand array
+// Using insertion sort because:
+// 1. Hand size is small (max 16 cards)
```

Comment:
```suggestion
// 1. Hand size is small
```

This is not correct, the number of cards is currently maxed at 8 but we plan to remove or increase that bound, the number should not appear here.